src/LodeRunner.API.Test/IntegrationTests/Extensions/HttpClientExtension.cs

```diff
@@ -382,5 +382,64 @@ public static async Task<HttpResponseMessage> DeleteItemById<TEntity>(this HttpC
 
             return httpResponse;
         }
+
+        /// <summary>
+        /// Gets the test run by identifier retries asynchronous.
+        /// </summary>
+        /// <typeparam name="TEntity">The Entity type.</typeparam>
+        /// <param name="httpClient">The httpClient.</param>
+        /// <param name="baseEntityUri">The endpoint Uri to the GetItem.</param>
+        /// <param name="itemId">The LoadTestConfig ID.</param>
+        /// <param name="jsonOptions">The json options.</param>
+        /// <param name="output">The output.</param>
+        /// <param name="validateCondition">validateCondition delegate.</param>
+        /// <param name="maxRetries">The maximum retries.</param>
+        /// <param name="timeBetweenRequestsMs">The time between requests ms.</param>
+        /// <returns>HttpStatusCode and TEntity from response.</returns>
+        public static async Task<(HttpStatusCode, TEntity)> GetEntityByIdRetries<TEntity>(this HttpClient httpClient, string baseEntityUri, string itemId, JsonSerializerOptions jsonOptions, ITestOutputHelper output, Func<TEntity, Task<(bool result, string fieldName, string conditionalValue)>> validateCondition, int maxRetries = 10, int timeBetweenRequestsMs = 1000)
+            where TEntity : BaseEntityModel
+        {
+            string entityName = typeof(TEntity).Name;
+
+            HttpResponseMessage httpResponse = new ();
+            TEntity testRun = null;
+
+            for (int i = 1; i <= maxRetries; i++)

```

Comment:
I see 3 maxRetries based loops.  It fill like you could reduce that to 
```csharp
RunAndRetry (Func<Task> taskToExecute, int maxRetries, int maxDelay) 
```
as a common way to re pass a function in and retry it with common code.  

I won't block for it, but if it'll take you an hour or less to update please do before merging.

Reply:
will do